src/test/scala/nl.knaw.dans.easy.validatebag/rules/profile/NumberedRulesSpec.scala

```diff
@@ -15,41 +15,36 @@
  */
 package nl.knaw.dans.easy.validatebag.rules.profile
 
-import java.net.URI
-import java.nio.charset.StandardCharsets
-
-import better.files.File
 import nl.knaw.dans.easy.validatebag._
-import nl.knaw.dans.easy.validatebag.rules.metadata.normalizeLicenseUri
-import nl.knaw.dans.lib.error._
-import org.apache.commons.configuration.PropertiesConfiguration
+import nl.knaw.dans.easy.validatebag.rules.{ ProfileVersion0, ProfileVersion1 }
+import org.scalatest.Inspectors
 
-class NumberedRulesSpec extends TestSupportFixture with CanConnectFixture {
+class NumberedRulesSpec extends TestSupportFixture with Inspectors {
 
-  private val resourceDir = File(getClass.getResource("/"))
-  private val configuration = new Configuration("version x.y.z",
-    new PropertiesConfiguration() {
-      setDelimiterParsingDisabled(true)
-      load(resourceDir / "debug-config" / "application.properties" toJava)
-    },
-    (resourceDir / "debug-config" / "licenses.txt")
-      .lines(StandardCharsets.UTF_8)
-      .filterNot(_.isEmpty)
-      .map(s => normalizeLicenseUri(new URI(s))).toSeq.collectResults.unsafeGetOrThrow
+  private val xmlValidators: Map[String, XmlValidator] = Map(
+    "dataset.xml" -> new XmlValidator(null),
+    "files.xml" -> new XmlValidator(null),
+    "agreements.xml" -> new XmlValidator(null)
   )
 
+  private val allRules: Map[ProfileVersion, RuleBase] = {
+    Map(
+      0 -> ProfileVersion0(xmlValidators, null),
+      1 -> ProfileVersion1(xmlValidators))
+  }
+
   "rulesCheck" should "succeed if all rules, that other rules depend on, exist" in {
     List(0, 1).map(version => {
-      val rulesBase = new EasyValidateDansBagApp(configuration).allRules(version)
-      val ruleNumbers = rulesBase.map(_.nr)
-      val result = rulesBase.flatMap(_.dependsOn).forall(ruleNumber => ruleNumbers.contains(ruleNumber))
-      result shouldBe true
+      val ruleNumbers = allRules(version).map(_.nr)
+      forEvery(allRules(version).flatMap(_.dependsOn)) {
+        dependency => ruleNumbers.contains(dependency) shouldBe true
```

Comment:
1. you can now also do `ruleNumbers should contain (dependency)` to get a better error message
2. you can apply the same trick with `forEvery` on the `List(0, 1).map`. Instead do
  `forEvery(0 to 1) { version => ... }`

Reply:
I modified the looping through the Profiles.  
But using `ruleNumbers should contain (dependency)` would, in contrary, give more confusing error messages.

Reply:
mmm, weird. I would guess that with your `shouldBe true`, you get a message like 'false was not equal to true'. With `should contain` I would expect to see a message like '[rules] did not contain [dependency]'.

When I check out your code and compare the two, I see a clear difference in clearness:

**ruleNumbers contains dependency shouldBe true**
```
forEvery failed, because: 
  at index 0, forEvery failed, because: 
    at index 2, false was not equal to true (NumberedRulesSpec.scala:43) 
  in List("1.2.1", "1.2.2(a)", "1.2.3(a)", "1.2.1", "1.2.4(a)", "1.2.1", "1.2.1", "1.2.1", "1.3.1(a)", "2.1", "2.1", "2.1", "2.2(a)", "3.1.1", "3.1.1", "3.1.1", "3.1.1", "3.1.1", "2.2(b)", "2.2(b)", "3.2.2", "3.2.3", "3.2.4", "3.2.2", "3.2.2") (NumberedRulesSpec.scala:40) 
in Range.Inclusive(0, 1)
```

**ruleNumbers should contain (dependency)**
```
forEvery failed, because: 
  at index 0, forEvery failed, because: 
    at index 2, List("1.1.1", "1.2.1", "1.2.2(a)", "1.2.2(b)", "1.2.3(b)", "1.2.4(a)", "1.2.4(b)", "1.2.5", "1.2.6(a)", "1.2.6(b)", "1.3.1(a)", "1.3.1(b)", "2.1", "2.2(a)", "2.2(b)", "2.5", "3.1.1", "3.1.2", "3.1.4", "3.1.5", "3.1.6", "3.1.7", "3.2.1", "3.2.2", "3.2.3", "3.2.4", "3.2.5", "3.2.6", "3.2.7", "3.3.1", "3.4.1") did not contain element "1.2.3(a)" (NumberedRulesSpec.scala:42) 
  in List("1.2.1", "1.2.2(a)", "1.2.3(a)", "1.2.1", "1.2.4(a)", "1.2.1", "1.2.1", "1.2.1", "1.3.1(a)", "2.1", "2.1", "2.1", "2.2(a)", "3.1.1", "3.1.1", "3.1.1", "3.1.1", "3.1.1", "2.2(b)", "2.2(b)", "3.2.2", "3.2.3", "3.2.4", "3.2.2", "3.2.2") (NumberedRulesSpec.scala:40) 
in Range.Inclusive(0, 1)
```

yes, the second one is longer, but contains much more information about the error at hand

Reply:
I find the second one confusing; 
`index 2` refers to the `List("1.2.1", "1.2.2(a)", "1.2.3(a)...` (dependencies list), but right after `index 2` you get another list, that of all NumbererRules.

Reply:
That's right, isn't it? `index 2` refers to the list below: `in List("1.2.1", "1.2.2(a)", "1.2.3(a)", "1.2.1", ...)`, where the 3rd item (index 2, start counting from 0) is the one that fails.